### DISCUSSION TITLE
Make Zed configurable to behave like an "omni" editor instead of "project-centric" editor

### DISCUSSION THREAD

**Discussion by verybigelephants** [zed-industries/zed#26349] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

First of all, i love Zed editor!

However, right now it is very "project -centric" and not being able to be used as a regular a swiss knife editor like for example Sublime - you open a workspace and are able to navigate files there. This is pretty awesome when you're developing  one single project.

But there are scenarios when you simply don't follow this workflow. You open multiple files in different folders, different servers, etc. and work on a different paradigm rather than developing a single project.

It would be awesome if we were able to configure Zed to be used as a regular text editor when we're not forced to have a single "project" opened and edit files in there, but rather to use it as a universal editing tool and switch between open file tabs as we please. 

Or am i missing something? Please forgive me my ignorance in case i have missed some program setting

### If applicable, add mockups / screenshots to help present your vision of the feature

n/a

**Comment by notpeter** [comment]
Perhaps I'm misunderstanding something but you can absolutely use zed as an editor of individual files without opening a specific folder as a project.

When I run the following: `zed ~/.zshrc ~/Brewfile ~/.yarnrc` I get a Zed window with three open tabs. If I subsequently run `zed ~/.ssh/config` I now have four open tabs.

Similarly, in my `~/.gitconfig` I have the following to spawn Zed for editing my commit messages (remove the `--new` if you want to reuse the existing window):
```
[core]
	editor = zed --new --wait
```

Are there specific missing features that makes Zed difficult to use in this mode for you?

Obviously it could be better:
- https://github.com/zed-industries/zed/issues/16378
- https://github.com/zed-industries/zed/issues/5229
- https://github.com/zed-industries/zed/issues/4887

**Comment by notpeter** [comment]
Or perhaps this is what you're going for:
- https://github.com/zed-industries/zed/issues/7136

**Comment by bilal-houari** [comment]
i think adding options for creating a file or open a folder without calling it a project would help. also by default zed always opens the most recent project and there is no clear way to close it and just have an interface where you can open files and folder.

i suggest adding something like the vscode welcome screen where there are the recently opened folders and options to create a new folder or file to work on.

![image](https://github.com/user-attachments/assets/645a98be-4527-4111-9d68-e499110899b5)

**Comment by dzuelke** [comment]
Note that the "there must be a project" assumption leads to other issues as well. For example, it's not possible to open a `/dev/fd/…` file using the `zed` CLI (in order to [work around the inability to pipe input to the `zed` command](https://github.com/zed-industries/zed/issues/4770#issuecomment-2258728884)) - that [causes an error in `worktree.rs`](https://github.com/zed-industries/zed/issues/4770#issuecomment-2378039654).

**Comment by notpeter** [comment]
> there is no clear way to close it and just have an interface where you can open files and folder.

I think some of what you're experiencing are things that have been lost in translation when compared to the MacOS experience.

1. `New Window` / `New File` actions are not exposed via the GUI menu on Linux (they are under the File Menu on Mac). You can use the command palette or keyboard shortcuts:

https://github.com/zed-industries/zed/blob/530286b31a71e5b806a17171801d0ef746b83994/assets/keymaps/default-linux.json#L363-L364

2. MacOS applications can exist without a window, specifically you can close the last window and the menu bar remains available.  As a result our behavior for workspace restoration on Mac is closer to what you might be expecting.  If you Quit Zed, we restore any open workspaces.  But if you close all Workspace windows and then Quit Zed, when you come back you are presented with an empty workspace with a single untitled buffer.

On Linux, we can't tell the difference between "Closing the last workspace window" and "Quitting Zed" because there's no way for Zed to be open without an associated Project window.  Currently when you quit zed, we always restore the last closed project when you return. Ironically, VSCode has this behavior on MacOS too -- the only way to open to an empty workspace (Welcome landing page) is to have close your project window, then open a new window with an empty workspace, then quit and you'll get the empty/welcome workspace start (same as Zed on Linux).

3. We currently have a welcome screen, but it's geared towards first open, not as a landing page for a seasoned user.  I think having a welcome page which focused on the returning user is a reasonable request and there has been previous design work on this.

---

I'm not sure the right answer as for how to differentiate between the user's intent to "close zed" (should restore project) or "close project" (should not restore) on Windows and Linux when spawned from the graphical environment.

A workaround in the meantime is to spawn zed from the cli with `zed -n`:
```
  -n, --new
          Create a new workspace
```

**Comment by jansol** [comment]
> I'm not sure the right answer as for how to differentiate between the user's intent to "close zed" (should restore project) or "close project" (should not restore) on Windows and Linux when spawned from the graphical environment.

My 2c:
* if zed was launched with one or more files (but no directories!) given as arguments, it's not a project and should not be saved in recent projects nor restored automatically
* if the arguments are directories, it should be treated as a project
* I'm a bit torn on what I'd expect to happen with a mix of files and directories as arguments
* when starting without any arguments I would expect to see the welcome screen (or the last project depending on settings)

**Comment by probably-neb** [comment]
Hello all, unfortunately much of Zed is built with the assumption that the user is working on a "project", so this feature would require a substantial amount of work to make this happen. Therefore, I'm going to convert this issue to a discussion, where the pro/cons of adding this functionality can be discussed further. Cheers!

**Comment by VKondakoff** [comment]
Just linking another one discussion on the same topic: [Show some love for the documents, which are not part of any project, please!](https://github.com/zed-industries/zed/discussions/24427)

**Comment by ShyamC-dev** [comment]
Ability to open a file outside of the current project is vital to switch from vscode. i can't have zed for project work and vsode for opening files. When working with a remote linux host, zed doesn't allow opening any file outside of the project. thats too bad,
